lcviewernoqt/drawitems/lcdimradial.cpp

```diff
@@ -34,21 +34,67 @@ void LCDimRadial::draw(LcPainter& painter, const LcDrawOptions &options, const l
     
     // Draw line
     EndCaps endCaps;
-    auto tLinePos = !mouseIsInside
-      ? mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), te.width + capSize)
-      : mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), -te.width - capSize);
-
-    auto tMText = !mouseIsInside
-      ? mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), te.width / 2 + capSize / 2)
-      : mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), -te.width / 2 - capSize / 2);
-      
-    painter.move_to(definitionPoint2().x(), definitionPoint2().y());
-    painter.line_to(mousePos.x(), mousePos.y());
-    painter.line_to(tLinePos.x(), mousePos.y());
-    painter.stroke();
-    endCaps.render(painter, EndCaps::CLOSEDARROW, mousePos.x(), mousePos.y(), definitionPoint2().x(), definitionPoint2().y(), capSize) ;
-
-    this->drawText(value, textAngle(), lc::TextConst::AttachmentPoint::Top_center, tMText, painter, options, rect);
+
+    const bool textisInside = middleOfText().distanceTo(mousePos) < radiusCircle;
+    const bool textFitsInside = radiusCircle * 2. > te.width;
+
+    if (textisInside && textFitsInside) {
+        auto tLinePos = (definitionPoint2().x() > mousePos.x())
+                ? lc::geo::Coordinate(mousePos.x() - (te.width + capSize), mousePos.y())
+                : lc::geo::Coordinate(mousePos.x() + (te.width + capSize), mousePos.y());
+        auto tMText = (definitionPoint2().x() > mousePos.x())
+                ? lc::geo::Coordinate(mousePos.x() - (te.width/2 + capSize/2), mousePos.y() - (te.width/4 + capSize/2))
+                : lc::geo::Coordinate(mousePos.x() + (te.width/2 + capSize/2), mousePos.y() - (te.width/4 + capSize/2));
+
+        if (definitionPoint2().y() <= mousePos.y()) {
+            tMText = (definitionPoint2().x() > mousePos.x())
+                    ? lc::geo::Coordinate(mousePos.x() - (te.width/2 + capSize/2), mousePos.y())
+                    : lc::geo::Coordinate(mousePos.x() + (te.width/2 + capSize/2), mousePos.y());
+        }
+
+        painter.move_to(definitionPoint2().x(), definitionPoint2().y());
+        painter.line_to(mousePos.x(), mousePos.y());
+        painter.line_to(tLinePos.x(), mousePos.y());
+        painter.stroke();
+        endCaps.render(painter, EndCaps::CLOSEDARROW, mousePos.x(), mousePos.y(), definitionPoint2().x(), definitionPoint2().y(), capSize) ;
+
```

Comment:
@bhattigurjot 
please make your own judgement call, and avoid get into the "deadlock" situation, as I explained early on: while one party is waiting for comments, and the other party is waiting to see actions.

Thanks!